extensions/spark/kyuubi-extension-spark-3-5/src/main/scala/org/apache/kyuubi/sql/MockOutputTable.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.sql
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.{CatalogTable, CatalogTableType, HiveTableRelation}
+import org.apache.spark.sql.catalyst.plans.logical.{InsertIntoStatement, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.util.CharVarcharUtils
+import org.apache.spark.sql.execution.command.DDLUtils
+import org.apache.spark.sql.execution.datasources.CreateTable
+
+import org.apache.kyuubi.sql.KyuubiSQLConf.{MOCK_OUTPUT_TABLE, MOCK_OUTPUT_TABLE_SUFFIX}
+
+case class MockOutputTable(spark: SparkSession) extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    if (!conf.getConf(MOCK_OUTPUT_TABLE)) {
+      plan
+    } else {
+      val suffix = conf.getConf(MOCK_OUTPUT_TABLE_SUFFIX).getOrElse(defaultMockSuffix)
+      plan resolveOperators {
+        // InsertInto
+        case i @ InsertIntoStatement(r: HiveTableRelation, _, _, _, _, _, _)
+            if !isMockTable(r.tableMeta) =>
+          val newTable = mockTable(r.tableMeta, suffix, true)
+          i.copy(table = DDLUtils.readHiveTable(newTable))
```

Comment:
We should think about more. For example, the mocked table is not existed, how can we do insert into ?